docs/usage/shoot_versions.md

```diff
@@ -0,0 +1,108 @@
+Using automatic upgrades however does not guarantee that a Shoot runs a non-deprecated version, as the latest version (overall or of the minor version) can be deprecated as well.
+Deprecated versions should have an expiration date set for eventual expiration. 
+
+- **expired:** An `expired` versions has an expiration date in the past. 
+ New clusters with that version cannot be created and existing clusters are forcefully migrated to a higher version during the maintenance time.
+ 
+Below is an example how the relevant section of the CloudProfile might look like:
+``` yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: CloudProfile
+metadata:
+  name: alicloud
+spec:
+  kubernetes:
+    versions:
+      - classification: supported
+        version: 1.17.1
+      - classification: deprecated
+        expirationDate: "2020-07-24T16:13:26Z"
+        version: 1.17.0
+      - classification: preview
+        version: 1.16.6
+      - classification: supported
+        version: 1.16.5
+      - classification: deprecated
+        expirationDate: "2020-04-25T09:30:40Z"
+        version: 1.16.4
+      - classification: supported
+        version: 1.15.7
+      - classification: deprecated
+        expirationDate: "2020-06-09T14:01:39Z"
+        version: 1.15.6
+```
+ 
+ ## Version Requirements (Kubernetes and Machine image)
+ 
+ The Gardener API server enforces the following requirements for versions:
+  
+ **Deletion of a version** 
+ - A version that is in use by a Shoot cannot be deleted from the CloudProfile.
+ 
+ **Adding a version** 
+ - A version must not have an expiration date in the past.
+ - There can be only one `supported` version per minor version.
+ 
+ ## Forceful migration of expired versions
+ 
+ If a Shoot is running a version after its expiration date has passed, it will be forcefully migrated during its maintenance time.
+ This happens **even if the owner has opted out of automatic cluster updates!**
+
+For **Machine images**, the Shoots worker pools will be updated to the latest `non-preview` version of the pools respective image.
+
+For **Kubernetes versions**, the forceful update picks the latest `non-preview` patch version of the current minor version. 
+
+ If the cluster is already on the latest patch version and the latest patch version is also expired, 
+ it will continue with the latest patch version of the **next consecutive minor Kubernetes version**, so **it will result in an 
+ update of a minor Kubernetes version!** 
+ 
+ If that's expired as well, the update process repeats until a non-expired Kubernetes version is reached.
+  
+ **Depending on the circumstances described above, it can happen that the cluster receives multiple consecutive minor Kubernetes version updates!**
+ As a Kubernetes version update cannot skip a minor version, the CloudProfile has to be maintained properly.
```

Comment:
What does "maintain properly" exactly mean? Also, can you make more clear that such minor version upgrades don't happen altogether (i.e., we don't go from 1.10 -> 1.11 -> 1.12 in one maintenance cycle, but instead do it in multiple cycles)?